File: services/python/models/post.py
from datetime import datetime
from typing import Optional

class Post:
    def __init__(
        self,
        title: str,
        content: str,
        user_id: int,
        id: Optional[int] = None,
        published: bool = False,
        created_at: Optional[datetime] = None,
        updated_at: Optional[datetime] = None
    ):
        self.id = id
        self.user_id = user_id
        self.title = title
        self.content = content
        self.published = published
        self.created_at = created_at or datetime.now()
        self.updated_at = updated_at or datetime.now()

    def publish(self):
        """Mark the post as published"""
        self.published = True
        self.updated_at = datetime.now()

    def unpublish(self):
        """Mark the post as unpublished"""
        self.published = False
        self.updated_at = datetime.now()

    def to_dict(self) -> dict:
        """Convert to dictionary"""
        return {
            'id': self.id,
            'user_id': self.user_id,
            'title': self.title,
            'content': self.content,
            'published': self.published,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'updated_at': self.updated_at.isoformat() if self.updated_at else None
        }

    def __repr__(self):
        return f"<Post(id={self.id}, title={self.title}, published={self.published})>"

File: services/rust/models/post.rs
use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Post {
    pub id: Option<i32>,
    pub user_id: i32,
    pub title: String,
    pub content: String,
    pub published: bool,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

impl Post {
    pub fn new(user_id: i32, title: String, content: String) -> Self {
        Self {
            id: None,
            user_id,
            title,
            content,
            published: false,
            created_at: Utc::now(),
            updated_at: Utc::now(),
        }
    }

    pub fn publish(&mut self) {
        self.published = true;
        self.updated_at = Utc::now();
    }

    pub fn unpublish(&mut self) {
        self.published = false;
        self.updated_at = Utc::now();
    }

    pub fn update(&mut self, title: Option<String>, content: Option<String>) {
        if let Some(t) = title {
            self.title = t;
        }
        if let Some(c) = content {
            self.content = c;
        }
        self.updated_at = Utc::now();
    }
}
